ckanext/ckanext-apply_permissions_for_service/ckanext/apply_permissions_for_service/model.py

```diff
@@ -85,7 +86,12 @@ def as_dict(self):
         application_dict['target_organization'] = toolkit.get_action('organization_show')(
             {'ignore_auth': True}, {'id': application_dict['target_organization_id']})
 
-        application_dict['intermediate_organization'] = toolkit.get_action('organization_show')(
-            {'ignore_auth': True}, {'id': application_dict['intermediate_organization_id']})
+        if "intermediate_organization_id" in application_dict \
+                and application_dict['intermediate_organization_id'] is not None:
```

Comment:
```suggestion
        if application_dict.get('intermediate_organization_id')
```

.get() call pretty much does the same thing but is more convenient.

Reply:
Fixed